### ISSUE TITLE
[Question] Lookup by exact IPA match

### ISSUE THREAD

**Issue by sevagh** [timmahrt/pysle#11]
Say I want to disambiguate two pronunciations of the word "bow". I can get these from your library (small excerpt from a large number of matches):
```
>>> isleDict.search('boʊ')
... ('bow', [('# b ˈoʊ #', ['nn', 'vb', 'vbp'])]), ('bow-and-quarter_line', [('# b ˈoʊ . n̩ # k w ˈɑ ɹ . t ə ɹ #', [])]),  ...
```
I'm having trouble figuring out how to modify the regex to only get `'# b ˈoʊ #'` without the extra stuff. Any suggestions? I tried the regular anchor characters (^, $) without luck.

**Comment by timmahrt** [comment]
You can narrow the search space by
```
isleDict.search('boʊ', wordInitial="only", wordFinal="only")
```

but this won't allow an exact match.  I'm wrapping the search string in some extra stuff, which is why ^ $ don't work, but maybe the wrapping isn't needed?

I'll need to dig a bit and will try to get back soon!

**Comment by timmahrt** [comment]
I've just released v2.3.0.  In this version, there is a new argument to search() `exactMatch`.  If True, it will only search for exact matches of the string input, ignoring stress, word boundaries, and syllables.

eg
```python
printOutMatches(u"kæt˺", exactMatch=True)
>> cat: # k ˌæ t˺ #()

printOutMatches(u"kæt˺")
>> cat: # k ˌæ t˺ #()
catatonia: # k ˌæ . t˺ ə . t ˈoʊ . n i . ə #()
catatonic: # k ˌæ . t˺ ə . t ˈɑ . n ɪ k #(jj)
```

Do you think it will work for your use case?

**Comment by sevagh** [comment]
Yes, this should work - except, now that I really think about my problem, I don't think it will help me achieve my goals.

What I really want is to look up the definition or sense of the word "bow, pronounced boʊ" vs. "bow, pronounced baʊ"

**Comment by timmahrt** [comment]
Yes, unfortunately this won't work for you then.  You'll need some resource that contains both the word, definition, and pronunciation.  Some online dictionaries have all of this information--actually now that I think about it wiktionary should have all of that info.

It would not be a small amount of work, but one could scrap all of the text in wiktionary, and for each word, get the pronunciation and the definition.
https://en.wiktionary.org/wiki/Help:FAQ#Downloading_Wiktionary

**Comment by sevagh** [comment]
Perfect, I've been looking for such a resource for a while - thanks for adding the exactMatch feature too.

**Comment by timmahrt** [comment]
Did you parse wiktionary yet?

I thought it was kindof a cool idea--and even not considering adding defitions, wiktionary has a lot more entries than the ISLEDict--so I tried to do it this weekend.  It wasn't as simple as I was hoping.  I've got a WIP up.  I think I might be busy for the next few weeks, so I'm not sure I'll be able to make progress on it.

https://github.com/timmahrt/wiktionaryParser/pull/1

Output looks like:
`abdomen,æbˈdəʊ.mən,noun,The fat surrounding the belly. <ref name=SOED></ref>`

The parser breaks on most pages and the valid output still needs to be cleaned up.

**Comment by sevagh** [comment]
I actually left it off as a "future work" part of the larger project - but that's awesome. I've seen a few mentions of mapping Wiktionary to WordNet in different NLP papers and projects so it should be relevant to other people.

For instance I found this: https://github.com/davidjurgens/crown/issues/3

**Comment by timmahrt** [comment]
I finished my preliminary work on this.  I suspect there are still some bugs that limit the results but I was able to do a data dump.  I've added the code and the dump to github
https://github.com/timmahrt/wiktionaryParser

Looking through the data, most words don't have pronunciations.  For those that do (~60K words out of ~1M), the pronunciation annotations are variable, which limit its usefulness to me--some words have syllables and stress marked but many don't.

If I pick this up again, I may investigate using some existing parsers that I missed my first time around ^^":
https://pypi.org/search/?q=wiktionaryparser

I will probably change my library name to avoid clashing with those
